src/scripts/create_arc.py

```diff
@@ -72,6 +80,72 @@ def main() -> None:
+            & (train_df.option_a.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+            & (train_df.option_b.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (train_df.option_b.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+            & (train_df.option_c.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (train_df.option_c.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+            & (train_df.option_d.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (train_df.option_d.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+        ]
+        val_df = val_df[
+            (val_df.instruction.str.len() >= MIN_NUM_CHARS_IN_INSTRUCTION)
+            & (val_df.instruction.str.len() <= MAX_NUM_CHARS_IN_INSTRUCTION)
+            & (val_df.option_a.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (val_df.option_a.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+            & (val_df.option_b.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (val_df.option_b.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+            & (val_df.option_c.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (val_df.option_c.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+            & (val_df.option_d.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (val_df.option_d.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+        ]
+        test_df = test_df[
+            (test_df.instruction.str.len() >= MIN_NUM_CHARS_IN_INSTRUCTION)
+            & (test_df.instruction.str.len() <= MAX_NUM_CHARS_IN_INSTRUCTION)
+            & (test_df.option_a.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (test_df.option_a.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+            & (test_df.option_b.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (test_df.option_b.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+            & (test_df.option_c.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (test_df.option_c.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+            & (test_df.option_d.str.len() >= MIN_NUM_CHARS_IN_OPTION)
+            & (test_df.option_d.str.len() <= MAX_NUM_CHARS_IN_OPTION)
+        ]
+
+        def is_repetitive(text: str) -> bool:
+            """Return True if the text is repetitive."""
+            max_repetitions = max(Counter(text.split()).values())
+            return max_repetitions > 50
+
+        # Remove overly repetitive samples
+        train_df = train_df[
+            ~train_df.instruction.apply(is_repetitive)
+            & ~train_df.option_a.apply(is_repetitive)
+            & ~train_df.option_b.apply(is_repetitive)
+            & ~train_df.option_c.apply(is_repetitive)
+            & ~train_df.option_d.apply(is_repetitive)
+        ]
+        val_df = val_df[
+            ~val_df.instruction.apply(is_repetitive)
+            & ~val_df.option_a.apply(is_repetitive)
+            & ~val_df.option_b.apply(is_repetitive)
+            & ~val_df.option_c.apply(is_repetitive)
+            & ~val_df.option_d.apply(is_repetitive)
+        ]
+        test_df = test_df[
```

Comment:
Maybe a small helper function that can be called for train_df, val_df, and test_df?